r/R/dplyr.R

```diff
@@ -631,17 +671,85 @@ abandon_ship <- function(call, .data, msg = NULL) {
   eval.parent(call, 2)
 }
 
-arrange.arrow_dplyr_query <- function(.data, ...) {
+arrange.arrow_dplyr_query <- function(.data, ..., .by_group = FALSE) {
+  call <- match.call()
+  exprs <- quos(...)
+  if (.by_group) {
+    exprs <- c(quos(!!!dplyr::groups(.data)), exprs)
+  }
+  if (length(exprs) == 0) {
+    # Nothing to do
+    return(.data)
+  }
   .data <- arrow_dplyr_query(.data)
   if (query_on_dataset(.data)) {
     not_implemented_for_dataset("arrange()")
   }
-  # TODO(ARROW-11703) move this to Arrow
-  call <- match.call()
-  abandon_ship(call, .data)
+  is_dataset <- query_on_dataset(.data)
+  if (is_dataset) {
```

Comment:
I believe this duplicates the block immediately above it. Also, unclear that you need to embargo datasets since you're only evaluating the sort on the Table after collecting().

Reply:
Oops. Dup code removed in ca38dba1069c323f279ba0affc98d743797d0ab9. I'll try removing this and testing on datasets.

Reply:
Implemented on datasets in 7f360d8eb16eb9b9876a77e2b122f78a23401956. Tests added in 320c3d56ea2225b6c3419ef01ee652c568678fb6.